cluster/ubuntu/util.sh

```diff
@@ -369,6 +374,15 @@ function provision-master() {
     ubuntu/binaries/master/ \
     "${MASTER}:~/kube"
 
+  if [ -z "$CNI_PLUGIN_CONF" ] || [ -z "$CNI_PLUGIN_EXES" ]; then
+    # Flannel is being used: copy the flannel binaries and scripts, set reconf flag
+    scp -r $SSH_OPTS ubuntu/master-flannel/* "${MASTER}:~/kube"
+    NEED_RECONFIG_DOCKER=true
+  else
+    # CNI is being used: set reconf flag
```

Comment:
Regarding tarball URL instead of file pathname set: the added generality is good.  Do you know a way to secure the `tar xf` command so that it doesn't write in unwanted places?

As for sanity checking, it seems a little inconsistent to check that the CNI_PLUGIN_EXES exist but not also check that the CNI_PLUGIN_CONF exists.

BTW, if we switch to one tarball URL for the executables then we could also get the CNI config file from the tarball, reducing two config envars to one.

BTW, switching from file pathname set to one tarball does not eliminate the need for checking; it imposes a need for certain structure in the tarball contents and checking for conformance to that.

Do we really need this much checking?  If so, can we do it as a later PR that builds on this one?